### DISCUSSION TITLE
Local setup for ArgoCD frontend dev?

### DISCUSSION THREAD

**Discussion by yeya24** [argoproj/argo-cd#8480] [Q&A] [answered]
Hi community, I am working on the ArgoCD web shell feature and I really want to know how to set up the frontend dev environment locally with a remote ArgoCD server backend.
My question is specific for:
1. How to configure the frontend config to connect to the remote ArgoCD?
2. We are using OIDC at the remote ArgoCD server. Is there a way to skip or disable it temporarily for local dev?

Thank you!

**Comment by alexmt** [comment, marked as answer]
Hello @yeya24 ,

I would suggest running API server and UI locally and connecting it to the remote Argo CD installation: 

* Make sure you have kubectl access to the remote K8S cluster that runs Argo CD
*  Port forward Argo CD redis:  `kubectl port-forward svc/argocd-redis-ha 6379:6379 -n argocd`
* Install goreman `go install github.com/mattn/goreman@latest` 
* Start API server and UI locally: `goreman start api-server ui`


API server will be started with disable authentication

**Reply by yeya24** [reply]
Thanks @alexmt for the detailed reply. I have one more question. If I have ArgoCD in the remote cluster and I start API server locally, then there are two API servers running, right? One at local and one at the remote cluster, is it fine?

**Reply by alexmt** [reply]
Right, two API servers should not cause any issues.
